lib/matplotlib/backends/__init__.py

```diff
@@ -47,12 +47,7 @@ def _get_running_interactive_framework():
                 if frame.f_code == tkinter.mainloop.__code__:
                     return "tk"
                 frame = frame.f_back
-    try:
-        from matplotlib.backends import _macosx
-    except ImportError:
-        pass
-    else:
-        if _macosx.event_loop_is_running():
+    if ('_macosx' in sys.modules) and (_macosx.event_loop_is_running()):
```

Comment:
You need to get the module under the correct name (`"matplotlib.backends._macosx" in sys.modules`) and then fetch it out of `sys.modules` (`sys.modules["matplotlib.backends._macosx"].event_loop_is_running()`) (can be split over multiple lines as for the other modules above) but other than that, it looks like a reasonable solution.